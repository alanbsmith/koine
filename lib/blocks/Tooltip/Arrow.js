import styled from 'styled-components';
import { rem } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

const modifierConfig = {
  bottom: (props) => `
    border-top-width: ${rem(props.theme.dimensions.basePad)};
    border-right-width: ${rem(props.theme.dimensions.basePad)};
    border-bottom-width: 0;
    border-left-width: ${rem(props.theme.dimensions.basePad)};
    border-top-color: ${props.theme.colors.base.chrome900};
    bottom: ${rem(-props.theme.dimensions.basePad + 1)};
  `,
  left: (props) => `
    border-top-width: ${rem(props.theme.dimensions.basePad)};
    border-right-width: ${rem(props.theme.dimensions.basePad)};
    border-bottom-width: ${rem(props.theme.dimensions.basePad)};
    border-left-width: 0;
    border-right-color: ${props.theme.colors.base.chrome900};
    left: ${rem(-props.theme.dimensions.basePad + 1)};
  `,
  right: (props) => `
    border-top-width: ${rem(props.theme.dimensions.basePad)};
    border-right-width: 0;
    border-bottom-width: ${rem(props.theme.dimensions.basePad)};
    border-left-width: ${rem(props.theme.dimensions.basePad)};
    border-left-color: ${props.theme.colors.base.chrome900};
    right: ${rem(-props.theme.dimensions.basePad + 1)};
  `,
  top: (props) => `
    border-top-width: 0;
    border-right-width: ${rem(props.theme.dimensions.basePad)};
    border-bottom-width: ${rem(props.theme.dimensions.basePad)};
    border-left-width: ${rem(props.theme.dimensions.basePad)};
    border-bottom-color: ${props.theme.colors.base.chrome900};
    top: ${rem(-props.theme.dimensions.basePad + 1)};
  `,
};

const Arrow = styled.div`
  border-color: transparent;
  border-style: solid;
  height: 0;
  position: absolute;
  width: 0;
  ${applyStyleModifiers(modifierConfig)}
`;

export default Arrow;
