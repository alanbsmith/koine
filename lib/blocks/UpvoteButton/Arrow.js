import React from 'react';
import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';
import { rem } from 'polished';
import FontAwesome from 'react-fontawesome';

function UnstyledArrow(props) {
  return (
    <FontAwesome
      name="arrow-up"
      {...props}
    />
  );
};

const modifierConfig = {
  orange: ({ theme }) => `color: ${theme.colors.base.orange};`,
  hoverBounce: ({ theme }) => `
    &:hover {
      animation: ${theme.animations.upvote} 1s infinite ease-in;
    }
  `,
};

const Arrow = styled(UnstyledArrow)`
  font-size: ${rem(24)};
  ${applyStyleModifiers(modifierConfig)}
`;

export default Arrow;
