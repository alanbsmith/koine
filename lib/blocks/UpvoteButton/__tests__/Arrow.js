import React from 'react';
import Arrow from '../Arrow';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Arrow Element', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Arrow />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the orange modifier', () => {
    const tree = renderer.create(
      <Arrow modifiers={['orange']}>A paragraph</Arrow>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the hoverBounce modifier', () => {
    const tree = renderer.create(
      <Arrow modifiers={['hoverBounce']}>A paragraph</Arrow>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
