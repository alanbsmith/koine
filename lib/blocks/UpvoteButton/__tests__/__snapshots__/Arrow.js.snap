// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow Element renders correctly 1`] = `
.c0 {
  font-size: 1.5rem;
}

<span
  aria-hidden={true}
  className="fa fa-arrow-up c0"
/>
`;
